lighttuner/scheduler/task_scheduler.py

```diff
@@ -170,12 +171,24 @@ def write_config_file(self, task_config_template_path: str, rl_config_file_path:
 
     def generate_config_file(self, task_config_template_path: str) -> List[str]:
         """generate config file as string list."""
+        main_entrance_pattern = re.compile(r'if__name__(==|is)(\"|\')__main__(\"|\'):')
```

Comment:
HAVE U EVER TESTED THESE LINES?
```python
if__name__ == '__main__':
if __name__ == '      __main__           ':
```

Please make your regular expression more accurate.

Reply:
Thanks for reminding me of the former case. Yes, it is a bug, and it is now handled by string check.
The latter could pass the test.

Reply:
But the latter one should not be pased by your check. The `__name__` asserted is `           __main__             ` instead of `__main__`, whcih means this is not runnable.

Reply:
Yes, you are right. It should be matched in complete order.